Source/UTF8LineParser.swift

```diff
@@ -58,16 +45,14 @@ class UTF8LineParser {
             case .emptyInput:
                 break Decode
             case .error(let len):
+                seenCr = false
                 if dataIter.position == data.endIndex {
                     // Error at end of block, carry over in case of split code point
                     remainderPos = data.index(data.endIndex, offsetBy: -len)
+                    // May as well break here as next will be .emptyInput
+                    break Decode
                 } else {
                     // Invalid character, replace with replacement character
-                    if seenCr {
```

Comment:
This is one place things are simpler because of immediately flushing on a carriage return. Because we've already flushed the line, we can just set `seenCr = false` above, since this invalid character cannot be line feed, rather than flush the previous line here.